roles/get-sitename/tasks/main.yml

```diff
@@ -83,12 +83,36 @@
 #     "url": "crceducation-dev.cardinalsites.stanford.edu"
 # },
 #
+# From something that looks like
+#  {
+#     "clayman.dev-cardinalsites.acsitefactory.com": {
+#     "name": "sbiybj5001706dev",
+#     "flags": [],
+#     "conf": {
+#       "gardens_site_id": 5001706,
+#       "gardens_db_name": "sbiybj5001706dev",
+#       "acsf_site_id": 5001706,
+#       "acsf_db_name": "sbiybj5001706dev",
+#       "site_api_key": "400b006fda61ada556d9e19171d4b88edef1fa82"
+#     }
+#   },
+#
+# The REST API has a restriction of 100 items per request and quickly falls over
+# for our needs. To bypass the API we are directly getting all of the site info
+# from sites.json on the acquia server. Unfortunately the initial format is not
+# usable and we need to parse out the machine name and id of the site factory
+# site by using the url for the sitefactory item as the machine name and getting
+# the key from the json array for the node id. We only use the acsitefactory uri
+# as it is the canonical machine name for the site and the other domains listed
+# cannot be trusted.
+#
 - name: Manipulate sites.json
   set_fact:
-    sites_formatted: "{{ sites_formatted|default([]) + [ {'url': item.0, 'id': item.1.acsf_site_id, 'site': item.0|regex_replace('((-dev|-stage|-test).cardinalsites.stanford.edu|.(dev-|stage-|test-)cardinalsites.acsitefactory.com|(-dev|-stage|-test).sites.stanford.edu)', '') } ] }}"
+    sites_formatted: "{{ sites_formatted|default([]) + [ {'url': item.0, 'id': item.1.acsf_site_id, 'site': item.0|regex_replace('.(dev-|stage-|test-)cardinalsites.acsitefactory.com', '') } ] }}"
```

Comment:
Why `dev-|stage-|test-`? `stage` is not used in URLs.

Also, I'm still not grokking how `sites_formatted` can reference itself in setting itself, nor am I entirely grokking how the order is `url`, `id`, `site`, in this task, but `id`, `site`, `url` in the output. Does one have to do with the other?

Reply:
It is a loop to build a dict.

On the first pass if it does not exist we instantiate it with an empty array `|default([])` then append the new array of values to it.  On the subsequent passes it gets more values appended.

Not sure why the debug puts it in alphabetical order. Probably pedantic ansible developers doing their best to keep it as tidy as possible.

Reply:
As the keys are object properties,order does not matter.

Reply:
Why dev, stage, and test? I always forget which of test and stage are not used and decided to cover all my butts.

Reply:
_Ahhhhh_, now I get it. Thanks; I'll add some code comments and remove the `stage-`

Have I ever told you how I feel about the term "stage"...?